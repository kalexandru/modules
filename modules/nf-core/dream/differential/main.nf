// TODO:
//      should this module take all variables already processed (meta),
//      or should it be able to take the yaml file and parse it on its own?

process DREAM_DIFFERENTIAL {
    tag "${meta.id} - ${meta.contrast_id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/d6/d6fa8a7908dd357484d63bb574a378f1739440a2c0752ce91b1e0e9d1ac1638c/data' :
        'community.wave.seqera.io/library/bioconductor-edger_bioconductor-variancepartition_r-optparse:ba778938d72f30c5' }"

    input:
    tuple val(meta), path(samplesheet), path(counts)

    output:
    tuple val(meta), path("*.dream.results.tsv")        , emit: results
    tuple val(meta), path("*.dream.mean_difference.png"), emit: md_plot
    tuple val(meta), path("*.MArrayMM.dream.rds")       , emit: rdata
    tuple val(meta), path("*.dream.model.txt")          , emit: model
    tuple val(meta), path('*.dream.contrasts_plot.png') , emit: contrasts_png
    tuple val(meta), path('*.dream.var_plot.png')       , emit: var_png
    tuple val(meta), path("*.R_sessionInfo.log")        , emit: session_info
    tuple val(meta), path("*.normalised_counts.tsv")    , emit: normalised_counts, optional: true
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix                 = task.ext.prefix ?: meta.id
    def blocking_factors       = meta.blocking_factors ? "--blocking_variables '${meta.blocking_factors.join(';')}'" : ''
    def exclude_samples_col    = meta.exclude_samples_col ? "--exclude_samples_col '${meta.exclude_samples_col.join(';')}'" : ''
    def exclude_samples_values = meta.exclude_samples_values ? "--exclude_samples_values '${meta.exclude_samples_values.join(';')}'" : ''
    def args                   = task.ext.args ?: ''
    """
    dream_de.R  \\
        --output_prefix ${meta.contrast_id} \\
        --count_file ${counts} \\
        --sample_file ${samplesheet} \\
        --contrast_variable ${meta.contrast_variable} \\
        --reference_level ${meta.contrast_reference} \\
        --target_level ${meta.contrast_target} \\
        ${blocking_factors} \\
        ${exclude_samples_col} \\
        ${exclude_samples_values} \\
        --threads ${task.cpus} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        r-optparse: \$(Rscript -e "library(optparse); cat(as.character(packageVersion('optparse')))")
        r-edger: \$(Rscript -e "library(edgeR); cat(as.character(packageVersion('edgeR')))")
        r-variancepartition: \$(Rscript -e "library(variancePartition); cat(as.character(packageVersion('variancePartition')))")
        r-biocparallel: \$(Rscript -e "library(BiocParallel); cat(as.character(packageVersion('BiocParallel')))")
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix   ?: "${meta.id}"
    """
    touch "${meta.contrast_id}.dream.results.tsv"
    touch "${meta.contrast_id}.dream.mean_difference.png"
    touch "${meta.contrast_id}.dream.contrasts_plot.png"
    touch "${meta.contrast_id}.dream.var_plot.png"
    touch "${meta.contrast_id}.MArrayMM.dream.rds"
    touch "${meta.contrast_id}.dream.model.txt"
    touch "${meta.contrast_id}.R_sessionInfo.log"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        r-optparse: \$(Rscript -e "library(optparse); cat(as.character(packageVersion('optparse')))")
        r-edger: \$(Rscript -e "library(edgeR); cat(as.character(packageVersion('edgeR')))")
        r-variancepartition: \$(Rscript -e "library(variancePartition); cat(as.character(packageVersion('variancePartition')))")
        r-piocparallel: \$(Rscript -e "library(BiocParallel); cat(as.character(packageVersion('BiocParallel')))")
    END_VERSIONS
    """
}
