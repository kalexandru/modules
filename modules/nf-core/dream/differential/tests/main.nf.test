nextflow_process {

    name "Test Process DREAM_DIFFERENTIAL"
    script "../main.nf"
    process "DREAM_DIFFERENTIAL"
    config "./nextflow.config"

    tag "modules"
    tag "models"
    tag "dge"
    tag "dream"
    tag "dream/differential"

    test("Mus musculus - expression table - contrasts") {

        when {
            process {
                """
                // contrast data + counts + samplesheet
                input[0] = Channel.of(
                    [
                        [
                            id: 'test',
                            contrast_id: 'treatment_mCherry_hND6',
                            contrast_variable: 'treatment',
                            contrast_reference: 'mCherry',
                            contrast_target: 'hND6',
                            blocking_factors: null
                        ],
                        file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/mus_musculus/rnaseq_expression/SRP254919.samplesheet.csv", checkIfExists: true),
                        file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("Mus musculus - expression table - contrasts + blocking factors") {

        when {
            process {
                """
                // contrast data + counts + samplesheet
                input[0] = Channel.of(
                    [
                        [
                            id: 'test-bf',
                            contrast_id: 'treatment_mCherry_hND6',
                            contrast_variable: 'treatment',
                            contrast_reference: 'mCherry',
                            contrast_target: 'hND6',
                            blocking_factors: ['sample_number']
                        ],
                        file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/mus_musculus/rnaseq_expression/SRP254919.samplesheet.csv", checkIfExists: true),
                        file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("Mus musculus - expression table - contrasts + blocking factors stub") {

        options '-stub'
        when {
            process {
                """
                // contrast data + counts + samplesheet
                input[0] = Channel.of(
                    [
                        [
                            id: 'test-stub',
                            contrast_id: 'treatment_mCherry_hND6',
                            contrast_variable: 'treatment',
                            contrast_reference: 'mCherry',
                            contrast_target: 'hND6',
                            blocking_factors: ['sample_number']
                        ],
                        file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/mus_musculus/rnaseq_expression/SRP254919.samplesheet.csv", checkIfExists: true),
                        file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/mus_musculus/rnaseq_expression/SRP254919.salmon.merged.gene_counts.top1000cov.tsv", checkIfExists: true)
                    ]
                )
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
