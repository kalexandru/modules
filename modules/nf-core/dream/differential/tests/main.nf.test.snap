{
    "Mus musculus - expression table - contrasts + blocking factors": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.results.tsv:md5,bd28cb586bb40ad9f0b7069cb3ada0a2"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.mean_difference.png:md5,0d1c235cca14dcf75af4ef6cb66247ae"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.MArrayMM.dream.rds:md5,7003d9bf41a8f4e9ba77be5fb733c2b7"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.model.txt:md5,17788b137dd6c2b2a8ed887aa7404f8d"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.contrasts_plot.png:md5,1fab14c3a7fb300bc6d8e693ef75cf27"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.var_plot.png:md5,d2b36eb27f2437140e97e47cf9fa243c"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.R_sessionInfo.log:md5,e591353ef3f5d63a74889d57f7ce37c6"
                    ]
                ],
                "7": [
                    
                ],
                "8": [
                    "versions.yml:md5,56f3e82916abb64aa39ae9d008d71978"
                ],
                "contrasts_png": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.contrasts_plot.png:md5,1fab14c3a7fb300bc6d8e693ef75cf27"
                    ]
                ],
                "md_plot": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.mean_difference.png:md5,0d1c235cca14dcf75af4ef6cb66247ae"
                    ]
                ],
                "model": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.model.txt:md5,17788b137dd6c2b2a8ed887aa7404f8d"
                    ]
                ],
                "normalised_counts": [
                    
                ],
                "rdata": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.MArrayMM.dream.rds:md5,7003d9bf41a8f4e9ba77be5fb733c2b7"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.results.tsv:md5,bd28cb586bb40ad9f0b7069cb3ada0a2"
                    ]
                ],
                "session_info": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.R_sessionInfo.log:md5,e591353ef3f5d63a74889d57f7ce37c6"
                    ]
                ],
                "var_png": [
                    [
                        {
                            "id": "test-bf",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.var_plot.png:md5,d2b36eb27f2437140e97e47cf9fa243c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,56f3e82916abb64aa39ae9d008d71978"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T22:35:42.737618467"
    },
    "Mus musculus - expression table - contrasts + blocking factors stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.mean_difference.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.MArrayMM.dream.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.model.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.contrasts_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.var_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.R_sessionInfo.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "7": [
                    
                ],
                "8": [
                    "versions.yml:md5,8338b523d0f508917db713c8a0cb7fdf"
                ],
                "contrasts_png": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.contrasts_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "md_plot": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.mean_difference.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "model": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.model.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "normalised_counts": [
                    
                ],
                "rdata": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.MArrayMM.dream.rds:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.results.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "session_info": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.R_sessionInfo.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "var_png": [
                    [
                        {
                            "id": "test-stub",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": [
                                "sample_number"
                            ]
                        },
                        "treatment_mCherry_hND6.dream.var_plot.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8338b523d0f508917db713c8a0cb7fdf"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T22:35:59.811030788"
    },
    "Mus musculus - expression table - contrasts": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.dream.results.tsv:md5,bd28cb586bb40ad9f0b7069cb3ada0a2"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.dream.mean_difference.png:md5,0d1c235cca14dcf75af4ef6cb66247ae"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.MArrayMM.dream.rds:md5,7003d9bf41a8f4e9ba77be5fb733c2b7"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.dream.model.txt:md5,17788b137dd6c2b2a8ed887aa7404f8d"
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.dream.contrasts_plot.png:md5,1fab14c3a7fb300bc6d8e693ef75cf27"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.dream.var_plot.png:md5,d2b36eb27f2437140e97e47cf9fa243c"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.R_sessionInfo.log:md5,e591353ef3f5d63a74889d57f7ce37c6"
                    ]
                ],
                "7": [
                    
                ],
                "8": [
                    "versions.yml:md5,56f3e82916abb64aa39ae9d008d71978"
                ],
                "contrasts_png": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.dream.contrasts_plot.png:md5,1fab14c3a7fb300bc6d8e693ef75cf27"
                    ]
                ],
                "md_plot": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.dream.mean_difference.png:md5,0d1c235cca14dcf75af4ef6cb66247ae"
                    ]
                ],
                "model": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.dream.model.txt:md5,17788b137dd6c2b2a8ed887aa7404f8d"
                    ]
                ],
                "normalised_counts": [
                    
                ],
                "rdata": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.MArrayMM.dream.rds:md5,7003d9bf41a8f4e9ba77be5fb733c2b7"
                    ]
                ],
                "results": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.dream.results.tsv:md5,bd28cb586bb40ad9f0b7069cb3ada0a2"
                    ]
                ],
                "session_info": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.R_sessionInfo.log:md5,e591353ef3f5d63a74889d57f7ce37c6"
                    ]
                ],
                "var_png": [
                    [
                        {
                            "id": "test",
                            "contrast_id": "treatment_mCherry_hND6",
                            "contrast_variable": "treatment",
                            "contrast_reference": "mCherry",
                            "contrast_target": "hND6",
                            "blocking_factors": null
                        },
                        "treatment_mCherry_hND6.dream.var_plot.png:md5,d2b36eb27f2437140e97e47cf9fa243c"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,56f3e82916abb64aa39ae9d008d71978"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T22:35:16.708141029"
    }
}